rust-arroyo/src/backends/kafka/types.rs

```diff
@@ -22,6 +23,14 @@ impl Headers {
         });
         Self { headers }
     }
+
+    pub fn find(self, key: &str) -> Option<Vec<u8>> {
+        if let Some(header) = self.headers.iter().find(|header| header.key == key) {
+            header.value.map(|v| v.to_vec())
+        } else {
+            None
+        }
+    }
```

Comment:
```suggestion
    pub fn get(self, key: &str) -> Option<&[u8]> {
        self.headers.iter().find(|header| header.key == key).and_then(|header| header.value.as_ref())
    }
```

I would suggest to name this `get`, as that is usually used for `HashMap`-like functions.
Also returning `&[u8]` there, as there shouldn’t be a reason to clone inside the function. The caller can do that if necessary.